### DISCUSSION TITLE
MIME type incorrectly detected as `text/html` since v3.13.0

### DISCUSSION THREAD

**Discussion by bakerkretzmar** [alpinejs/alpine#3811] [7. Bugs (archived)]
Since Alpine v3.13.0, files containing Alpine's code are often automatically detected as having a MIME type of `text/html`.

**Reproduction**

- Download https://unpkg.com/alpinejs@3.12.3/dist/cdn.min.js and https://unpkg.com/alpinejs@3.13.0/dist/cdn.min.js.
- Run `file --mime-type path-to-alpine-3.12.3`—output is `text/plain`.
- Run `file --mime-type path-to-alpine-3.13.0`—output is `text/html`.

**Background**

It appears that this happens when the MIME type of a file containing Alpine is detected using the file's _contents_. Creating a file object/stream (in PHP), or of course using the file extension, works fine. I'm pretty sure this is because in Alpine v3.13 the string `<body>` appears much closer to the beginning of the file than it used to:

v3.12.3:

<img width="2298" alt="image" src="https://github.com/alpinejs/alpine/assets/18192441/1e8c9496-e7b4-4275-85d6-8ba092ee9f78">

v3.13.0:

<img width="2297" alt="image" src="https://github.com/alpinejs/alpine/assets/18192441/36018679-9967-4a81-8373-19dbd83945a8">
<br/>
<br/>

Is there any way to control or escape that string so the file isn't detected as HTML?

**Comment by SimoTod** [comment]
Do you have a use case where this is causing an issue? If it's a problem and that is the reason, an obvious fix would be to write `body` instead of `<body>` since it's just a warning message.

**Reply by bakerkretzmar** [reply]
Not sure it's a _necessary_ use case, but what I was doing was using Laravel's `Storage` Facade to upload built assets to DigitalOcean Spaces. I was loading the contents of the files as strings and uploading those, so PHP detected the buffered string as HTML. Switching to using a PHP stream resource of the actual file fixed it (because then it just uses the file extension).

I can see this continuing to cause issues in situations where a MIME type is being detected from a string, and Alpine's code happens to be what's first in that string. Not sure how common that is, but since it happens with the `file` CLI tool and PHP's `finfo_buffer()` function I feel like it could come up again.

**Comment by ekwoka** [comment]
You should not rely on automatic mime detection.

Your server should have no issues seeing that what is being requested should be JS and sending back the appropriate mime type regardless of the actual contents.
